framework/src/problems/FEProblemBase.C

```diff
@@ -3693,6 +3693,53 @@ FEProblemBase::setCurrentExecuteOnFlag(const ExecFlagType & flag)
   _current_execute_on_flag = flag;
 }
 
+void
+FEProblemBase::executeAllObjects(const ExecFlagType & exec_type)
+{
+  // Set the current flag
+  setCurrentExecuteOnFlag(exec_type);
+  if (exec_type == EXEC_NONLINEAR)
+  {
+    _currently_computing_jacobian = true;
+    if (_displaced_problem)
+      _displaced_problem->setCurrentlyComputingJacobian(true);
+  }
+
+  // Samplers; EXEC_INITIAL is not called because the Sampler::init() method that is called after
+  // construction makes the first Sampler::execute() call. This ensures that the random number
+  // generator object is the correct state prior to any other object (e.g., Transfers) attempts to
+  // extract data from the Sampler. That is, if the Sampler::execute() call is delayed to here
+  // then it is not in the correct state for other objects.
+  if (exec_type != EXEC_INITIAL)
+    executeSamplers(exec_type);
+
+  computeUserObjects(exec_type, Moose::PRE_AUX);
+
+  // TODO: this triggers execute warehouse errors when there are no objects
+  // for the dynamic/custom runner execute on flags.  Figure out how to
+  // fix this.
+  // computeAuxiliaryKernels(exec_type);
+
+  computeUserObjects(exec_type, Moose::POST_AUX);
+
+  if (exec_type != EXEC_INITIAL)
+  {
+    // Pre-ic UserObjects should execute along with the Post-aux group except on EXEC_INITIAL
+    computeUserObjects(exec_type, Moose::PRE_IC);
+    if (_control_warehouse.hasExecType(exec_type))
+      executeControls(exec_type);
+  }
+
+  // Return the current flag to None
+  setCurrentExecuteOnFlag(EXEC_NONE);
+  _currently_computing_jacobian = false;
+  if (_displaced_problem)
+    _displaced_problem->setCurrentlyComputingJacobian(false);
+
+  if (_control_warehouse.hasExecType(exec_type))
+    executeControls(exec_type);
```

Comment:
`executeControls` again?

Reply:
This function is 90% just the same as `execute()` below.  Are you sure this is needed?  Can we at least call `execute()` for most of it?

Reply:
The things that execute does actually cause errors when called from a runner context - because they assume some setup/stuff that doesn't occur outside the regular executioner stack.  I'm actually open to just making this function be empty - i.e. to be implemented in the future.  Because it's already quite incomplete in terms of what we would want for the new runner system.

Reply:
Well - I think it's useful - even if incomplete.  Can you just add a boolean to `execute` to turn off the things you don't want to run right now?  Just trying to keep down duplication.